### PR TITLE
Tower Logic

### PR DESCRIPTION
**Pull request by e1pupper**
Added a base logic for towers in sprites.py and added a temporary asset for tower.

### CODE REVIEW

File: src/tower/sprites.py [hunk 1]
```diff
@@ -38,48 +38,42 @@ def proj_speed(self, level: int) -> float:
...
-    ):
+        radius: int = 100
+    ):  # IDK if scale is important but it's in the docs https://github.com/e1pupper/tower
         """Tower constructor"""
         super().__init__(name, desc, scale)
 
         self.level = level
+        self.radius = radius
+
+        self.sprite_path = Path(__file__).parent.parent.parent / "assets" / "sprites" / "towers" / f"{name}.png"
```
Comment (henri-climber) [inline src/tower/sprites.py:73]: use ASSETS_DIR constant please

File: src/tower/sprites.py [hunk 2]
```diff
@@ -38,48 +38,42 @@ def proj_speed(self, level: int) -> float:
...
 
     def __init__(self, name: str, desc: str, speed: float, scale: int = 1):
         """Enemy constructor"""
-        super().__init__(name, desc, scale)
 
+        self.name = name
+        self.desc = desc
         self.speed = speed
+
+        self.sprite_path = Path(__file__).parent.parent.parent / "assets" / "sprites" / "enemies" / f"{name}.png"
```
Comment (henri-climber) [inline src/tower/sprites.py]: same for this one

**Review by henri-climber — CHANGES_REQUESTED** [review]

File: src/tower/sprites.py (hunk 2 above)
Comment (e1pupper) [reply src/tower/sprites.py]: my fault brodie

File: src/tower/sprites.py [hunk 3]
```diff
@@ -37,49 +37,53 @@ def proj_speed(self, level: int) -> float:
...
+        name: str, 
+        desc: str, 
+        level: int, 
+        attacks: list[AttackSpec], 
         scale: int = 1,
-    ):
+        radius: int = 100
+    ):  # IDK if scale is important but it's in the docs https://github.com/e1pupper/tower
         """Tower constructor"""
         super().__init__(name, desc, scale)
```
Comment (FieryIceStickie) [inline src/tower/sprites.py:68]: this line should be removed if we're ditching Entity

File: src/tower/game_view.py [hunk 4]
```diff
@@ -31,6 +32,12 @@ def setup(self):
         self.scene = arcade.Scene.from_tilemap(self.tile_map)
 
         self.towers = arcade.SpriteList()
+        
+        self.towers.append(sprites.Tower("tower1", "A basic tower", 1, [sprites.AttackSpec("Basic Attack", "A basic attack", 1, 1, 1)]))
```
Comment (FieryIceStickie) [inline src/tower/game_view.py:36]: This is a placeholder so it's fine for now but for future Tower instantiations I think it'd be more readable to do it on multiple lines, especially the AttackSpec

**Review by FieryIceStickie — CHANGES_REQUESTED** [review]

File: src/tower/game_view.py (hunk 4 above)
Comment (henri-climber) [reply src/tower/game_view.py:36]: lint will change it anyways, line is way too long :D

**Comment by henri-climber** [comment]
This PR was merges with #21

**Pull request closed without merging**